tests/end2end/fixtures/qutewm.py

```diff
@@ -0,0 +1,133 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2016 Daniel Schadt
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Fixtures for the qutewm window manager."""
+
+import os
+import re
+import sys
+
+import pytest
+from PyQt5.QtCore import pyqtSignal
+
+from end2end.fixtures import testprocess
+
+
+class QuteWMProcess(testprocess.Process):
+
+    """Abstraction over a running qutewm instance.
+
+    Signals:
+        window_opened: Emitted when a window is opened.
+        window_closed: Emitted when a window is closed.
+        window_focused: Emitted when a window is focused.
+    """
+
+    SCRIPT = 'qutewm_sub'
+
+    window_opened = pyqtSignal(int)
+    window_closed = pyqtSignal(int)
+    window_focused = pyqtSignal(int)
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self.wm_failed = False
+        self._event_re = re.compile(
+            r'window (created|closed|activated): \[0x([0-9a-f]+)\]$')
+
+    def _parse_line(self, line):
+        self._log(line)
+        if 'event loop started' in line:
+            self.ready.emit()
+        elif 'Another window manager is running, exiting' in line:
+            self.wm_failed = True
+            self.ready.emit()
+        else:
+            match = self._event_re.search(line)
+            if match:
+                signal = {
+                    'created': self.window_opened,
+                    'closed': self.window_closed,
+                    'activated': self.window_focused,
+                }[match.group(1)]
+                signal.emit(int(match.group(2), 16))
```

Comment:
You mean something like `'signal: activated'`  is parsed and when `'signal: foobar'` is encountered we raise an exception ("Invalid signal from qutewm")?